### PR TITLE
ci: updates to ci and clean up of unused profiles (DNA-16592: DNA-17262)

### PR DESCRIPTION
**Pull request by cverhoef**
Can be merge to main as it is unrelated to framework updates:
- New ci pipeline versions
- SQL fluff support new version
- Cleanup unused profiles

### CODE REVIEW

File: .pipelines/azure-pipelines-transformations.yml [hunk 1]
```diff
@@ -16,81 +13,107 @@ parameters:
...
-# Therefore it is not necessary to also start it via a PR.
-pr:
+# Trigger the pipeline for all PRs we open, but do not start as long as there is no PR.
+trigger:
   none
 
+pr:
+  branches:
+    include:
+    - '*'
```
Comment (JasperSelmanUiPath) [inline .pipelines/azure-pipelines-transformations.yml:41]: We explicitly had set this to only run on a trigger. This means in practice that for each push we check the transformations for partners. We did this so we did not force partners to work with PRs (at least until they want to merge). 

Personally I am fine with changing this and require from our partners to work with PRs for their work, but if we do not want this we should not change this part.

File: README.md
Comment (JasperSelmanUiPath) [inline README.md:1]: I see we missed in previous updates to update this README. There is currently only 1 `dbt_project.yml` left, but in the README we refer on several places to two different projects.

**Review by JasperSelmanUiPath — CHANGES_REQUESTED** [review]

**Comment by JasperSelmanUiPath** [comment]
- The SQL contains linting issues.
- We should also update the dbt_project number or set it to the same placeholder we have in our repository.

File: .pipelines/azure-pipelines-transformations.yml (hunk 1 above)
Comment (cverhoef) [reply .pipelines/azure-pipelines-transformations.yml:41]: Will put it back to the original state.

**Review by JasperSelmanUiPath — APPROVED** [review]

**Pull request merged**